### ISSUE TITLE
ESLint lints the web client's build output: the tsup runner bundle isn't ignored, so the only lint warning comes from vendored undici

### ISSUE THREAD

**Issue by cliffhall** [modelcontextprotocol/inspector#2043]
## The problem

`clients/web/eslint.config.js` ignores `dist`, `storybook-static` and `coverage` — but not `build`:

```js
globalIgnores(["dist", "storybook-static", "coverage"]),
```

`clients/web/build/` is the tsup **runner bundle** (the prod Node web server, ~1.2 MB of generated JS including a vendored `undici`). Because it isn't ignored, `npm run lint` in `clients/web` lints generated third-party code, and the only lint output the web client produces today comes from inside that bundle:

```
clients/web/build/undici-HXPKCIY3.js
  16021:11  warning  Unused eslint-disable directive (no problems were reported from 'no-fallthrough')

✖ 1 problem (0 errors, 1 warning)
```

That directive is undici's own, in undici's own source. There is nothing to fix in this repo, and nothing anyone can act on.

## Why web is the odd one out

The other three clients already ignore their build output. Web is the only client that emits **both** `dist` (the Vite SPA) and `build` (the tsup runner), and it looks like only the first was carried into the ignore list:

| Client | `globalIgnores` | Emits `build/`? |
| --- | --- | --- |
| `clients/web` | `dist`, `storybook-static`, `coverage` | **yes — not ignored** |
| `clients/cli` | `build`, `coverage` | yes — ignored |
| `clients/tui` | `build`, `coverage` | yes — ignored |
| `clients/launcher` | `build`, `coverage` | yes — ignored |

The root config is consistent with the three: it already ignores `core/**/build/**`, `core/**/dist/**` and `test-servers/build/**`.

## Why it's worth fixing rather than tolerating

It's a warning today, so `validate` still passes — but:

- **It trains people to ignore lint output.** The web client's lint is never clean, and the one line it prints is noise. A real warning added later lands in a channel everyone has learned to skim past.
- **A rule promotion turns generated code into a gate failure.** `reportUnusedDisableDirectives` is a warning by default; anything that raises it — or any new rule that a bundled dependency happens to trip — fails `validate` on a file nobody wrote. #1959 (enabling `no-floating-promises` across all five scopes) is exactly the kind of change that walks into this.
- **It's wasted work.** Every `npm run lint` parses ~1.2 MB of bundled output, on a path that is gitignored and rebuilt on every `npm run build`.
- **Nothing guards it.** `verify:format-coverage` and `verify:typecheck-coverage` assert that first-party source is *covered*; there is no guard asserting that generated output is *excluded*, so this drifted silently and would again.

## The fix

Add `build` to the web client's ignore list, matching its three siblings:

```diff
-  globalIgnores(["dist", "storybook-static", "coverage"]),
+  globalIgnores(["build", "dist", "storybook-static", "coverage"]),
```

Verified locally: with that one line, `npx eslint .` in `clients/web` prints nothing and exits 0; without it, it prints the undici warning above.

While in there, it's worth a quick check of whether the same omission exists anywhere else a bundler writes into a linted scope.

## Acceptance

- `cd clients/web && npm run lint` produces **no output** and exits 0 on a tree that has been built.
- `npm run validate` from the root stays green.
- No file under any `clients/*/build` or `clients/*/dist` is reported by ESLint from any scope.

## Notes

Found while smoke-testing the v2.3.0 milestone. Pre-existing — not a regression introduced by anything in this milestone.

**Comment by cliffhall** [comment]
**Triage:** Priority **Medium** (total 6)

- Severity 2 — minor friction with an easy workaround (the warning can be skimmed past; nothing is broken)
- Urgency 3 — wanted this milestone (scheduled into v2.3.0 deliberately)
- Bonuses: +1 milestoned at creation (approved through the documented create flow)

Board: #28, Status `Todo`, milestone `v2.3.0`.
